app/src/main/java/by/khryshchanovich/androidapplication/light_sensor/MainSensorActivity.kt

```diff
@@ -0,0 +1,65 @@
+package by.khryshchanovich.androidapplication.light_sensor
+
+import android.content.Context
+import android.hardware.Sensor
+import android.hardware.SensorEvent
+import android.hardware.SensorEventListener
+import android.hardware.SensorManager
+import android.os.Build
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.annotation.RequiresApi
+import androidx.recyclerview.widget.LinearLayoutManager
+import by.khryshchanovich.androidapplication.R
+import kotlinx.android.synthetic.main.activity_main_sensor.*
+
+class MainSensorActivity : AppCompatActivity(), SensorEventListener {
+
+    lateinit var sensorManager: SensorManager
+    lateinit var sensorLight: Sensor
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main_sensor)
+
+        sensorManager = getSystemService(Context.SENSOR_SERVICE) as SensorManager
+        sensorLight = sensorManager.getDefaultSensor(Sensor.TYPE_LIGHT)
+    }
+
+    override fun onResume() {
+        super.onResume()
+        sensorManager.registerListener(this, sensorLight, SensorManager.SENSOR_DELAY_NORMAL)
+    }
+
+    override fun onPause() {
+        super.onPause()
+        sensorManager.unregisterListener(this)
+    }
+
+    override fun onAccuracyChanged(sensor: Sensor?, accuracy: Int) {
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    override fun onSensorChanged(event: SensorEvent?) {
+        if (event!!.sensor.type == Sensor.TYPE_LIGHT) {
+            getLightSensor(event)
+        }
+    }
+
+    @RequiresApi(Build.VERSION_CODES.LOLLIPOP)
+    fun getLightSensor(event: SensorEvent?) {
+        sensor_recycler.layoutManager = LinearLayoutManager(this)
+        val light = ArrayList<LightSensor>()
+        light.add(LightSensor("Light level, lx ", event!!.values[0].toString()))
```

Comment:
not use assert calls(!!) it is bad practice

Reply:
Thanks. I've fixed that bug.